pandas/io/sql.py

```diff
@@ -1258,10 +1255,31 @@ class PandasSQL(PandasObject):
     """
 
     def read_sql(self, *args, **kwargs):
-        raise ValueError(
-            "PandasSQL must be created with an SQLAlchemy "
-            "connectable or sqlite connection"
-        )
+        raise AbstractMethodError(self)
```

Comment:
`read_sql` actually doesn't get defined in either subclass - I think it could be deleted. Also, `read_table` is only defined in `SQLDatabase`, not `SQLiteDatabase`, so I don't think `@abstractmethod` would work for it.  `AbstractMethodError` is being used in `BaseEngine`, so I borrowed this for `PandasSQL`.

Reply:
Yeah I think `AbstractMethodError` is a legacy exception class from when pandas supported Python 2 and 3, so using `NotImplementedError` or `@abstractmethod` is probably preferred

Reply:
I changed `AbstractMethodError` to a combination of `NotImplementedError` and `@abstractmethod`.